### PR TITLE
(PXP-5597) Export to Seven Bridges

### PR DESCRIPTION
**Pull request by mpingram**
Adds Export to Seven Bridges button to explorer, which allows user to export a PFB to Seven Bridges' analysis platform. This feature is configured with the `export-to-seven-bridges` button type and the `sevenBridgesExportURL` parameter in the portal config. 

### New Features
- Added Export to Seven Bridges button, allowing user to export a PFB to Seven Bridges.

### Dependency updates
- Adds new button type `export-to-seven-bridges` to portal config `dataExplorerConfig.buttons.` Example:
```
      {
        "enabled": true,
        "type": "export-to-seven-bridges",
        "title": "Export All to Seven Bridges",
        "rightIcon": "external-link"
      },
``` 
- Adds new parameter `dataExplorerConfig.sevenBridgesExportURL` to portal config, which specifies the URL belonging to Seven Bridges that portal should redirect to. Portal appends the signed url to this URL with a query param (`&PFB=<signed url>`). Example:
```
 "sevenBridgesExportURL": "https://windmill.sbgenomics.com/import/windmill"
```

### CODE REVIEW

File: src/GuppyDataExplorer/index.jsx [hunk 1]
```diff
@@ -91,6 +91,7 @@ class Explorer extends React.Component {
               dropdowns: explorerConfig[this.state.tab].dropdowns,
               terraExportURL: explorerConfig[this.state.tab].terraExportURL,
               terraTemplate: explorerConfig[this.state.tab].terraTemplate,
+              sevenBridgesExportURL: explorerConfig[this.state.tab].sevenBridgesExportURL,
```
Comment (mfshao) [inline src/GuppyDataExplorer/index.jsx:94]: i'm start think manybe we should check if these optional value exists before enabling corresponding buttons, but might be out of the scope of your ticket 🤔

Comment (mpingram) [reply src/GuppyDataExplorer/index.jsx:94]: That makes sense -- I'll see if there's a simple way I can add that. What do you think of adding a check when portal reads the config, so portal fails to build if the config has the terra / sbg export buttons but is missing the `sevenBridgesExportURL` / `terraExportURL` parameters

File: src/GuppyDataExplorer/ExplorerButtonGroup/index.jsx [hunk 2]
```diff
@@ -449,7 +473,7 @@ class ExplorerButtonGroup extends React.Component {
     if (buttonConfig.type === 'export-to-pfb') {
       return !this.state.exportingToCloud;
     }
-    if (buttonConfig.type === 'export') {
+    if (buttonConfig.type === 'export' || buttonConfig.type === 'export-to-seven-bridges') {
       // if exportingToCloud is true or the PFB job is running, the button is not enabled.
       return !(this.state.exportingToCloud || this.isPFBRunning());
```
[the remark this answers is not in the document]
Comment (mpingram) [reply src/GuppyDataExplorer/ExplorerButtonGroup/index.jsx:478]: Good catch, thank you!

File: src/GuppyDataExplorer/ExplorerButtonGroup/index.jsx [hunk 3]
```diff
@@ -49,7 +50,15 @@ class ExplorerButtonGroup extends React.Component {
...
               exportingToCloud: false,
             }, () => {
-              this.sendPFBToCloud();
+              this.sendPFBToTerra();
+            });
+          } else if (this.state.exportingToSevenBridges) {
+            this.setState({
+              exportPFBURL: `${res.data.output}`.split('\n'),
+              toasterOpen: false,
+              exportingToCloud: false,
```
[the remark this answers is not in the document]
Comment (mpingram) [reply src/GuppyDataExplorer/ExplorerButtonGroup/index.jsx:59]: Yes, another good catch. Thank you Mingfei!

File: src/GuppyDataExplorer/index.jsx (hunk 1 above)
Comment (mfshao) [reply src/GuppyDataExplorer/index.jsx:94]: fails to build sounds a little bit too harsh to me. Personly I think throwing an error in the console would be enough. Even better if we can disable the button

Comment (mpingram) [reply src/GuppyDataExplorer/index.jsx:94]: Gotchu. My reasoning with fail to build was that we would catch configuration errors during our PR checks. I'll plan on adding an error check to this code, logging an error to the console, and disabling the button.

Comment (mfshao) [reply src/GuppyDataExplorer/index.jsx:94]: yeah i'm totally with you on that. We have the sanity-check.js which can bail out if sth is not right. But we also have places only throws errors. I guees both are acceptable, just the first option sounds more work on your plate to me 😛

Comment (mpingram) [reply src/GuppyDataExplorer/index.jsx:94]: Haha that sounds good. Thanks again for the R!

File: src/GuppyDataExplorer/ExplorerButtonGroup/index.jsx [hunk 4]
```diff
@@ -447,11 +471,30 @@ class ExplorerButtonGroup extends React.Component {
       return this.state.manifestEntryCount > 0;
     }
     if (buttonConfig.type === 'export-to-pfb') {
-      return !this.state.exportingToCloud;
+      // disable the pfb export button if any other pfb export jobs are running
+      return !(this.state.exportingToTerra || this.state.exportingToSevenBridges);
```
Comment (mfshao) [inline src/GuppyDataExplorer/ExplorerButtonGroup/index.jsx:475]: shoudl we also check for `isPFBRunning` here? too many button types 🤯

Comment (mpingram) [reply src/GuppyDataExplorer/ExplorerButtonGroup/index.jsx:475]: Haha yeah, I find this code hard to follow. I'll be honest that I don't fully understand it, but the original code didn't check for `isPFBRunning` and as far as I can tell the `isPending` check for export-to-pfb uses `isPFBRunning() == true` to decide if the button is pending (https://github.com/uc-cdis/data-portal/pull/686/files/67fe2b01dd812ce6a929815eebb95f403beaf83d#diff-a6734b941e5a6867b94a16ee82e40e96L471). So I thought it might have been intentional, so that the button is pending but not disabled when isPFBRunning() == true and we're not exporting to Terra (or seven bridges)

Comment (mfshao) [reply src/GuppyDataExplorer/ExplorerButtonGroup/index.jsx:475]: yeah it was not in there in the first place, i'm ok with that 👍
